clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java

```diff
@@ -158,6 +158,8 @@ public void testValidators() {
         testValidators(Type.STRING, ValidString.in("good", "values", "default"), "default",
                 new Object[]{"good", "values", "default"}, new Object[]{"bad", "inputs", null});
         testValidators(Type.LIST, ConfigDef.ValidList.in("1", "2", "3"), "1", new Object[]{"1", "2", "3"}, new Object[]{"4", "5", "6"});
+        testValidators(Type.STRING, ConfigDef.NON_NULL, "a", new Object[]{"abb"}, new Object[] {null});
+        testValidators(Type.STRING, ConfigDef.Composite.of(ConfigDef.NON_NULL, ValidString.in("a", "b")), "a", new Object[]{"a", "b"}, new Object[] {null, -1, "c"});
```

Comment:
@hachikuji aren't these tests you are asking for? Do you want me to add others as `AbstractConfigTest`?

Reply:
Ha, you are right. Glossed right over this.